pipeline_manager/src/main.rs

```diff
@@ -823,13 +1023,25 @@ pub(self) struct KillPipelineRequest {
 ///
 /// The pipeline is not deleted from the database, but its
 /// `killed` flag is set to `true`.
-///
-/// # HTTP errors
-///
-/// * `NOT_FOUND` - `pipeline_id` does not exist in the database.
-///
-/// * `INTERNAL_SERVER_ERROR` - pipeline process returned an HTTP
-/// error in response to the termination command.
+#[utoipa::path(
+    request_body = KillPipelineRequest,
+    responses(
+        (status = OK
+            , description = "Pipeline successfully terminated."
+            , body = String
+            , example = json!("Pipeline successfully terminated")
+            , example = json!("Pipeline already killed")),
+        (status = NOT_FOUND
+            , description = "Specified `pipeline_id` does not exist in the database."
+            , body = ErrorResponse
+            , example = json!(ErrorResponse::new("Unknown pipeline id '64'"))),
+        (status = INTERNAL_SERVER_ERROR
+            , description = "Request failed."
+            , body = ErrorResponse
+            , example = json!(ErrorResponse::new("Failed to kill the pipeline; response from pipeline controller: ..."))),
+    ),
+    tag = "Pipeline"
+)]
 #[post("/kill_pipeline")]
```

Comment:
`DELETE /pipeline/{pipeline_id}` (or POST?) (what's the difference between kill pipeline and delete? is it just stop vs delete?)

Reply:
delete = kill + delete from DB.  A killed pipeline can still be inspected until it is deleted.

Reply:
For pipelines, it makes sense to have a status field for the pipeline that we can change with a `PUT` or a `PATCH` to start or stop it. So we'll have:

```
CREATE /pipelines
PUT/PATCH /pipelines/{pipeline_id}
DELETE /pipelines/{pipeline_id}
```